plutus-use-cases/src/Language/PlutusTx/Coordination/Contracts/Escrow.hs

```diff
@@ -234,23 +228,25 @@ payEp
     -> Contract s e TxId
 payEp escrow = do
     vl <- endpoint @"pay-escrow"
-    pay escrow vl
+    pay (scriptInstance escrow) escrow vl
 
 -- | Pay some money into the escrow contract.
 pay
     :: ( HasWriteTx s
        , HasOwnPubKey s
        , AsContractError e
        )
-    => EscrowParams DataValue
+    => ScriptInstance Escrow
+    -- ^ The instance
+    -> EscrowParams DataValue
```

Comment:
I do find it a bit gross to be passing both the params and the instance derived from them, but I'm not sure what else we can do here.

Reply:
I can add a type that combines the two, but I don't think there's a way around passing the two of them around together if we want to avoid computing the instance over and over again (which causes the slowdown)